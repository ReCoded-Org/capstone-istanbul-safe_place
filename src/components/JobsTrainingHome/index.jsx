import React, { useState } from "react";
import { Container, Row, Col } from "react-bootstrap";
import "./index.scss";
import { useTranslation } from "react-i18next";
import OpportunitiesCard from "../OpportunitiesCard";

export default function JobsTrainingHome() {
  const { t } = useTranslation();
  const [type, setType] = useState("jobOpportunities");
  const cardsArr = [...t(`home.${type}`, { returnObjects: true })];
  const showTwoOpportunitiesCards = cardsArr.slice(0,2)
  const cards = showTwoOpportunitiesCards.map((opportunity) => (
      <OpportunitiesCard {...opportunity} key={opportunity.id} />
    ));
  return (
    <div className="jobsTrainingSection">
      <h2 className="title">Jobs / Trainings Opportunities</h2>
      <p className="description">
        Browse our list of job opportunities, trainings or workshops.{" "}
      </p>
      <Container>
        <Row>
          <Col md={3}>
            <div>
              <button
                type="button"
                className="jobsBtn"
                onClick={() => setType("jobOpportunities")}
              >
                Job Opportunities
              </button>
            </div>
            <div>
              <button
                type="button"
                className="trainingsBtn"
                onClick={() => setType("trainings")}
              >
                Trainings
              </button>
            </div>
          </Col>
          <Col md={9}>
            <Row>{cards}</Row>
          </Col>
        </Row>
      </Container>
    </div>
  );
}
