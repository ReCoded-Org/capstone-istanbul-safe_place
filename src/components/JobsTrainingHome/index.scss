@import "../../style/variables";
@import "../../style/buttons";

.jobsTrainingSection {
  padding-top: 3.125rem;
  margin-top: 1.875rem;
  padding-bottom: 3.125rem;
  background-color: $secondaryColor;
  width: 100%;
  text-align: center;
  font-family: $mainFontFamily;
}

.title {
  color: $primaryColor;
  font-size: 1.75rem;
}

.trainingsBtn,
.jobsBtn {
  @include primaryBtn;
  width: 12.75rem;
  border-radius: 0;
  background-color: $secondaryColor;

  &:hover {
    border-right: solid $primaryColor;
    color: $primaryColor;
  }

  &:focus {
    border-right: solid $primaryColor;
    outline: 0;
    color: $primaryColor;
  }
}

.jobsBtn {
  margin-top: 1.875rem;
}
