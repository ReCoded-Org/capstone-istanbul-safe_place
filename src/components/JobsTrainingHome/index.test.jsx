import React from "react";
import renderer from "react-test-renderer";
import JobsTrainingHome from "./index";

it("renders correctly and matches the snapshot", () => {
  const tree = renderer.create(<JobsTrainingHome />).toJSON();
  expect(tree).toMatchSnapshot();
});
