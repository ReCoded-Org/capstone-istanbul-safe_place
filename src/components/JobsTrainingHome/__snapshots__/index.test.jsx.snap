// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches the snapshot 1`] = `
<div
  className="jobsTrainingSection"
>
  <div
    className="jobsTrainingHeader"
  >
    <h2>
      jobsTrainingPage.header
    </h2>
    <h6>
      jobsTrainingPage.subHeader
    </h6>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-3"
      >
        <div>
          <button
            className="opportunityListBtn"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            jobsTrainingPage.buttons.jobOpportunities
          </button>
          <button
            className="opportunityListBtn"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            jobsTrainingPage.buttons.trainings
          </button>
        </div>
      </div>
      <div
        className="col-md-9"
      >
        <div
          className="row"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <a
      href="/opportunities"
      onClick={[Function]}
    >
      <h6
        className="opportunitiesButton"
      >
        home.buttons.opportunitiesBtn
         
        <span>
          â†’
        </span>
      </h6>
    </a>
  </div>
</div>
`;
